Development.md

```diff
@@ -0,0 +1,248 @@
+
+Production contracts can be deployed & upgraded with the production deployer `0x7C94E07bbf73518B0E25D1Be200a5b58F46F9dC7` (Private Key required)
+The current sepolia infrastructure is deployed from `xai-foundation/sentry:master`
+
+### Public Node
+
+The public node functions as an additional source of truth, its a xai-mainnet node that will publish Xai assertion block-hashes to a google cloud bucket CDN.
+It consists of 3 docker containers:
+
+- Xai Mainnet node used to read assertions from
+- Arbitrum-One Nitro node used for the parent layer RPC for the xai node
+- Publisher, a javascript runtime that reads the xai-node container logs and posts assertions data to the CDN
+
+The public node can be setup on any host with enough resources to run two arbitrum nodes. The host setup can be done through terraform. The configs needed are in `infrastructure\nitro-node-wrapper`, the publisher javascript runtime at `infrastructure\nitro-node-wrapper\publisher\index.mjs` 
+
+Specific environment variables have to be provided in a `.env`:
+```
+# Arb1 Node Settings
+# The Etehreum mainnet RPC to sync the ARB1 node with (prob > 100K requests to sync)
+ETH_RPC_URL= 
+
+# GBUCKET Project ID
+PROJECT_ID=
+# GBUCKET Account Email
+SERVICE_ACCOUNT_EMAIL=
+# GBUCKET PRIVATE KEY
+SERVICE_ACCOUNT_PRIVATE_KEY=
+# GBUCKET BUCKET NAME
+BUCKET_NAME=xai-sentry-public-node
+```
+
+Using the XAI Quicknode ETH RPC - Quicknode access required
+Google Cloud Storage `xai-sentry-public-node` access required
+
+Assertions currently get published to the Google Cloud Bucket https://console.cloud.google.com/storage/browser/xai-sentry-public-node/assertions
+
+Currently the public node is setup at https://console.cloud.google.com/compute/instancesDetail/zones/us-east4-b/instances/arbitrum-full-node-us?hl=de&project=xai-watchtower-node (Google Cloud access required for project `xai-watchtower-node`)
+
+**Manage the containers** (needs user to be in docker group or use `sudo`)
+- Stop all containers gracefully, in `/opt/public-node/build` `docker stop --time=300 $(docker ps -aq) && docker compose down`
+- Restart public-node-stack, in `/opt/public-node/build` `docker compose up -d` (requires .env in directory)
+- View logs, in `/opt/public-node/build` `docker compose logs -f -n 100` to view the last 100 logs and follow
+
+The xai node and arb node data will be stored in `/opt/public-node/data`
+
+
+# Deployed Runtimes
+
+## Challenger
+
+The sentry CLI running `boot-challenger` on a google cloud compute VM instance 
+https://console.cloud.google.com/compute/instances?hl=de&project=xai-sentry-node-challenger
+
+Google Cloud access to project `xai-sentry-node-challenger` required
+
+Running in a `screen` session running the CLI on the VM.
+Needs the "challenger" privateKey (`0xc74c1e08963ceef0e1f2f2a2eeb879f443e86836`) and bls privateKey (`0x977f9fe1d570ba7e48ea76566718023b260ed56eba368ff20dca06e99e650804126663c1a5febdb065ef139a0156a9ad`) on startup
```

Comment:
Same as above, I understand this runtime is deployed, but why is it here and what does it do?